cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1019,12 +1019,13 @@ if(ARROW_USE_BOOST)
     # Find static boost headers and libs
     set(Boost_USE_STATIC_LIBS ON)
   endif()
+  if(ARROW_BOOST_REQUIRE_LIBRARY)
+    set(ARROW_BOOST_COMPONENTS "COMPONENTS;system;filesystem")
+  endif()
   resolve_dependency(Boost
                      REQUIRED_VERSION
                      ${ARROW_BOOST_REQUIRED_VERSION}
-                     COMPONENTS
-                     system
-                     filesystem
+                     ${ARROW_BOOST_COMPONENTS}
```

Comment:
We can specify empty `COMPONENTS` like #13846.

Reply:
What is the advantage of your approach? The `COMPONENTS` arg completely missing or being empty has the same effect as `if(ARG_COMPONENTS)` will be false.

Reply:
(also sorry for just opening a PR for a ticket you are assigned  to. This issue was preventing us from submitting the R package to CRAN so we needed a fix fast and wanted to merge it back now)

Reply:
Yes. Both approaches have the same effect.
I think that the empty `COMPONENTS` approach is easiar to read. Putting the `COMPONENTS` keyword by a variable expansion may be a bit tricky.

If you like your approach, could you use `ARROW_BOOST_COMPONENTS_ARGS` or something instead of `ARROW_BOOST_COMPONENTS` for the variable name? Because the `COMPONENTS` keyword isn't a component.

Reply:
Like @kou I would find `COMPONENTS ${ARROW_BOOST_COMPONENTS}` to be slightly more explicit.

Reply:
Ok than let's go with @kou s version in #13846